graylog2-server/src/main/java/org/graylog/plugins/sidecar/services/CollectorService.java

```diff
@@ -101,7 +102,8 @@ public Collector fromRequest(Collector request) {
                 request.executablePath(),
                 request.executeParameters(),
                 request.validationParameters(),
-                request.defaultTemplate());
+                request.defaultTemplate(),
+                request.defaultTemplate() == null ? null : Collector.checksum(request.defaultTemplate().getBytes(StandardCharsets.UTF_8)));
```

Comment:
I think this won't work if there are changes in the default template, because now you would save the CRC from the changed template. If you want to do it without frontend changes, you have to load the existing collector and take the CRC from it.

Reply:
Yeah - quick and dirty is not a good idea. I actually encountered this myself earlier. Will make a more considered chante.

Reply:
The FE shouldn't need to know anything about CRC. I added the required logic to the REST APIs.

Reply:
I think we can always set defaultTemplateCRC to null here. Otherwise it's a bit confusing why we always calculate a new CRC.